src/System.Private.CoreLib/shared/System/Threading/ThreadPool.cs

```diff
@@ -1010,6 +1012,8 @@ public static partial class ThreadPool
         {
             if (millisecondsTimeOutInterval < -1)
                 throw new ArgumentOutOfRangeException(nameof(millisecondsTimeOutInterval), SR.ArgumentOutOfRange_NeedNonNegOrNegative1);
+            if (millisecondsTimeOutInterval > (uint)int.MaxValue)
```

Comment:
Could you please submit a test to CoreFX repo to cover this condition?

Reply:
Also, it may be nice to submit this bugfix in separate CoreCLR repo PR since it is pretty unrelated to the rest of the portable threadpool  changes.

Reply:
Yep, will do both. I expected this PR to be very short-lived so I just bolted the change on, but I underestimated the dependencies.

Reply:
I would be happy to merge the PR that just moves the files over with minor adjustments while you are working through the dependencies.

Reply:
It doesn't make sense to merge it without the CoreFX dependency. It won't work on Mono anyway. And if I update the Unix Interop files it will break CoreRT.